bindings/python/soletta/tests/_soletta_tests.pyx

```diff
@@ -0,0 +1,77 @@
+import soletta.asyncio
+from libc.stdint cimport uint32_t
+from cython.operator cimport dereference as dref
+
+cdef extern from "sol-mainloop.h":
+	struct sol_timeout:
+		pass
+
+	sol_timeout* sol_timeout_add(uint32_t, bint (*)(void*), void*)
+	bint sol_timeout_del(void*)
+
+
+tests = {}
+
+def add_test(func):
+	tests[func] = False
+
+cdef bint cb(void* data):
+	cdef bint success = dref(<bint*>data)
+	success = True
+	return False
+
+cdef bint cb2(void* data):
+		cdef int* d = <int*>data
+		cdef int count = dref(d)
+		count = count + 1
+
+		return True
+
+def test_timeout():
+	success = False
+	t = sol_timeout_add(100, cb, <void*>&success)
+
+	def post():
+		return success
+	
+
+def test_timeout_del():
+	cdef sol_timeout* t
+
+	cdef int callbackcounts = 0
+
+	t = sol_timeout_add(100, cb2, <void*>&callbackcounts)
+
+	def test_timeout_del_post():
+		return callbackcounts == 1
+	
+	return test_timeout_del_post
+
+def run_tests():
+
+	import asyncio
+	
```

Comment:
I miss a test that deletes the `sol_timeout_del()` from the callback and returns `False`, that should not double-free.

And of course review that returning true is running the CB, it seems not given `callbackcounts == 1` above. My hint there is from `cb2` you count and if ` == expected: return False`, this way you let it run for a while and see what happens.

It is also missing multiple tasks running at the same time.